src/group/mls_group/apply_commit.rs

```diff
@@ -127,7 +127,7 @@ impl MlsGroup {
         let own_confirmation_tag = ConfirmationTag::new(
             &ciphersuite,
             &provisional_epoch_secrets.confirmation_key,
-            &confirmed_transcript_hash,
+            &Secret::from(confirmed_transcript_hash),
```

Comment:
The transcript hashes are not secrets.

Reply:
I agree. However, I have changed the `extract` and `expand` functions to expect `Secret`s as inputs for both `salt` and `ikm`, so I need to make it a secret here for the moment. Note that this particular instance will go away once the confirmation tag is a MAC.

Reply:
I have changed it now such that the `ConfirmationTag` constructor takes `&[u8]` as input and the transformation to `Secret` is handled internally. It's indeed a bit nicer now. Was that what you meant?